homeassistant/components/huawei_lte/config_flow.py

```diff
@@ -0,0 +1,204 @@
+        """Handle user initiated config flow."""
+        if user_input is None:
+            return await self._async_show_user_form()
+
+        errors = {}
+
+        # Normalize URL
+        user_input[CONF_URL] = url_normalize(
+            user_input[CONF_URL], default_scheme="http"
+        )
+        if "://" not in user_input[CONF_URL]:
+            errors[CONF_URL] = "invalid_url"
+            return await self._async_show_user_form(
+                user_input=user_input, errors=errors
+            )
+
+        # See if we already have a router configured with this URL
+        existing_urls = [  # existing entries
+            url_normalize(entry.data[CONF_URL], default_scheme="http")
+            for entry in self._async_current_entries()
+        ]
+        if DOMAIN in self.hass.data:
+            existing_urls.extend(  # yaml configs
+                url_normalize(x, default_scheme="http")
+                for x in self.hass.data[DOMAIN].routers
+            )
+        if user_input[CONF_URL] in existing_urls:
+            return self.async_abort(reason="already_configured")
+
+        conn = None
+
+        def logout():
+            if hasattr(conn, "user"):
+                try:
+                    conn.user.logout()
+                except BaseException:
+                    _LOGGER.debug("Could not logout", exc_info=True)
+
+        username = user_input.get(CONF_USERNAME)
+        password = user_input.get(CONF_PASSWORD)
+        try:
+            if username or password:
+                conn = AuthorizedConnection(
+                    user_input[CONF_URL], username=username, password=password
+                )
+            else:
+                try:
+                    conn = AuthorizedConnection(
+                        user_input[CONF_URL], username="", password=""
+                    )
+                    user_input[CONF_USERNAME] = ""
+                    user_input[CONF_PASSWORD] = ""
+                except ResponseErrorException:
+                    _LOGGER.debug(
+                        "Could not login with empty credentials, proceeding unauthenticated",
+                        exc_info=True,
+                    )
+                    conn = Connection(user_input[CONF_URL])
+                    del user_input[CONF_USERNAME]
+                    del user_input[CONF_PASSWORD]
+        except LoginErrorUsernameWrongException:
+            errors[CONF_USERNAME] = "incorrect_username"
+        except LoginErrorPasswordWrongException:
+            errors[CONF_PASSWORD] = "incorrect_password"
+        except LoginErrorUsernamePasswordWrongException:
+            errors[CONF_USERNAME] = "incorrect_username_or_password"
+        except LoginErrorUsernamePasswordOverrunException:
+            errors["base"] = "login_attempts_exceeded"
+        except ResponseErrorException:
+            _LOGGER.warning("Response error", exc_info=True)
+            errors["base"] = "response_error"
+        except BaseException:
```

Comment:
Don't except `BaseException`. If necessary except `Exception` for unknown errors and disable broad except pylint warning.

Preferably handle the connection error specifically.

Reply:
Why would catching Exception and messing with pylint settings be preferable to just catching BaseException?

The API doesn't specify what the possible connection errors or exceptions are, so I don't think it's appropriate to dig out and use an implementation detail like that.

Reply:
`BaseException` is broader in scope than `Exception`. We don't except or use `BaseException` in the source except in a couple of modules where a 3rd party library uses that.

Reply:
I can see that, however it's pretty much a matter of taste and deserves documentation/reasoning beyond just "we don't do that" so that people don't need to waste time discussing it in reviews. For example in https://developers.home-assistant.io/docs/en/next/development_guidelines.html#use-new-style-string-formatting

Anyway, changed.